mbed-hal/spi_api.h

```diff
@@ -119,45 +177,34 @@ int  spi_master_write(spi_t *obj, int value);
 /** Checks if the specified SPI peripheral is in use
  *
  * @param[in] obj The SPI peripheral to check
- * @return non-zero if the peripheral is currently transmitting
+ * @return  rc  `SPI_RC_NOT_BUSY` or `SPI_RC_BUSY`
  */
-int  spi_busy(spi_t *obj);
-
-/** Get the module number
- *
- * @param[in] obj The SPI peripheral to check
- * @return The module number
- */
-uint8_t spi_get_module(spi_t *obj);
+spi_return_code_t  spi_busy(spi_t *obj);
 
 /**@}*/
 
-#if DEVICE_SPI_ASYNCH
 /**
  * \defgroup AsynchSPI Asynchronous SPI Hardware Abstraction Layer
  * @{
  */
 
-/** Begin the SPI transfer. Buffer pointers and lengths are specified in tx_buff and rx_buff
+/** Begin the SPI transfer.
+ * Buffer pointers and lengths are specified in the `spi_t *obj` object!
  *
  * @param[in] obj       The SPI object which holds the transfer information
- * @param[in] tx        The buffer to send
- * @param[in] tx_length The number of words to transmit
- * @param[in] rx        The buffer to receive
- * @param[in] rx_length The number of words to receive
- * @param[in] bit_width The bit width of buffer words
- * @param[in] event     The logical OR of events to be registered
  * @param[in] handler   SPI interrupt handler
- * @param[in] hint      A suggestion for how to use DMA with this transfer
+ *
+ * @return  rc  `SPI_RC_OK` or `SPI_RC_TRANSFER_LENGTH_TOO_LARGE`
  */
-void spi_master_transfer(spi_t *obj, void *tx, size_t tx_length, void *rx, size_t rx_length, uint32_t handler, uint32_t event, DMAUsage hint);
+spi_return_code_t spi_master_transfer(spi_t *obj, uint32_t handler);
 
 /** The asynchronous IRQ handler
  *
  * Reads the received values out of the RX FIFO, writes values into the TX FIFO and checks for transfer termination
  * conditions, such as buffer overflows or transfer complete.
- * @param[in] obj     The SPI object which holds the transfer information
- * @return event flags if a transfer termination condition was met or 0 otherwise.
+ *
+ * @param[in] obj    The SPI object which holds the transfer information
+ * @return event     SPI event flags, zero if no event
  */
 uint32_t spi_irq_handler_asynch(spi_t *obj);
```

Comment:
+1  - we need that common layer. Can you state that in the return description - all the events that occurred